darts/utils/data/horizon_based_dataset.py

```diff
@@ -107,7 +117,7 @@ def __getitem__(self, idx: int) -> Tuple[np.ndarray, Optional[np.ndarray], np.nd
 
         # optionally, load covariates
         ts_covariate = self.covariates[ts_idx] if self.covariates is not None else None
-        cov_type = CovariateType.NONE if self.covariates is None else CovariateType.PAST
+        main_cov_type = CovariateType.NONE if self.covariates is None else CovariateType.PAST
```

Comment:
Shouldn't you re-use `self.covariate_type` (and set it to None in init() if covariates are None)?